src/humancellatlas/data/metadata/helpers/staging_area.py

```diff
@@ -0,0 +1,330 @@
+
+    @property
+    def to_manifest_entry(self):
+        """
+        The content of a descriptor transformed into a format ready to create a
+        ManifestEntry object.
+        """
+        return {
+            'content-type': self.content['content_type'],
+            'crc32c': self.content['crc32c'],
+            'indexed': False,
+            'name': self.content['file_name'],
+            's3_etag': self.content['s3_etag'],
+            'sha1': self.content['sha1'],
+            'sha256': self.content['sha256'],
+            'size': self.content['size'],
+            'uuid': str(uuid5(self.namespace, self.content['file_name'])),
+            'version': self.content['file_version']
+        }
+
+
+@attr.s(frozen=True, kw_only=True, auto_attribs=True)
+class StagingArea:
+    links: Mapping[str, LinksFile]  # Key is the subgraph ID aka links_id
+    metadata: Mapping[str, MetadataFile]  # Key is the entity ID
+    descriptors: Mapping[str, DescriptorFile]  # Key is the entity ID
+
+    def get_bundle(self, subgraph_id: str) -> Tuple[str, List[JSON], JSON]:
+        """
+        Return a tuple consisting of the version of the downloaded bundle, a
+        list of the manifest entries for all metadata files in the bundle, and a
+        dictionary mapping the file name of each metadata file in the bundle to
+        the JSON contents of that file.
+        """
+        logger.debug('Composing bundle %s', subgraph_id)
+
+        links_file = self.links[subgraph_id]
+        manifest = []
+        metadata = {
+            'links.json': links_file.content
+        }
+        entity_ids_by_type = self._entity_ids_by_type(subgraph_id)
+        for entity_type, entity_ids in entity_ids_by_type.items():
+            # Sort entity_ids to produce the same ordering on multiple runs
+            for i, entity_id in enumerate(sorted(entity_ids)):
+                json_file_name = f'{entity_type}_{i}.json'
+                metadata_file = self.metadata[entity_id]
+                json_content = metadata_file.content
+                metadata[json_file_name] = json_content
+                file_manifest = {
+                    'content-type': 'application/json;',
+                    'crc32c': '0' * 8,
+                    'indexed': True,
+                    'name': json_file_name,
+                    's3_etag': None,
+                    'sha1': None,
+                    'sha256': '0' * 64,
+                    'size': len(json.dumps(json_content)),
+                    'uuid': metadata_file.uuid,
+                    'version': metadata_file.version
+                }
+                manifest.append(file_manifest)
+                if entity_type.endswith('_file'):
+                    file_manifest = self.descriptors[entity_id].to_manifest_entry
+                    manifest.append(file_manifest)
+                else:
+                    pass
+        logger.debug('Bundle contents: %s', list(metadata.keys()))
```

Comment:
Either 

```suggestion
        if( logger.isEnabledFor(DEBUG): logger.debug('Bundle contents: %r', list(metadata.keys()))
```

or 
```suggestion
```

I don't really see the point in only listing the keys and listing everything is definitely too much for a log messages, eve a debug one.

Reply:
Listing the keys told us which metadata files were included in the bundle, which seemed to be useful debug information.  I'd like to have some kind of debug message at the completion of  `get_bundle`  to pair with the debug message `'Composing bundle %s', subgraph_id` at the start.
How about just a count of files like `'Composed bundle with %i metadata files', len(metadata)`